### PR TITLE
TYP: Annotate plotting stacker

### PR DESCRIPTION
**Pull request by jbrockmendel**
In AreaPlot._plot we call `ax.fill_between`, which means `ax` must be an `Axes` object (and in particular, not an `Axis` object).  This chases down all the other places we can infer `Axes` from that.

Then some edits in an `__init__` to get mypy passing, and revert annotations of `_plot` in a few places because mypy complained about signature mismatch.

### CODE REVIEW

File: pandas/plotting/_matplotlib/boxplot.py
Comment (simonjayhawkins) [inline pandas/plotting/_matplotlib/boxplot.py:157]: I've opened #36034 so that when reviewing we can  just click on and get definition as doesn't work for string literals (in VS Code anyway)

File: pandas/plotting/_matplotlib/core.py
Comment (simonjayhawkins) [inline pandas/plotting/_matplotlib/core.py:668]: if adding ignores can you add the mypy error code and the mypy error message as a comment.

In this case, it's a relative simple cleanup. add a variable type declaration for args?

File: pandas/plotting/_matplotlib/core.py
Comment (simonjayhawkins) [inline pandas/plotting/_matplotlib/core.py:917]: possibly a use of Literal after vendoring PR merged

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @jbrockmendel lgtm except 'bare' #type: ignore

**Review by simonjayhawkins — APPROVED** [review]
Thanks @jbrockmendel lgtm pending green

**Comment by WillAyd** [comment]
yea also lgtm test failures seems unrelated so restarted. merge on green

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**
